src/CoreWCF.BuildTools/src/OperationParameterInjectionGenerator.Emitter.cs

```diff
@@ -51,12 +53,14 @@ public void Decrement()
             }
 
             private readonly OperationParameterInjectionSourceGenerationContext _sourceGenerationContext;
-            private SourceGenerationSpec _generationSpec;
+            private readonly SourceGenerationSpec _generationSpec;
 
-            public Emitter(in OperationParameterInjectionSourceGenerationContext sourceGenerationContext, SourceGenerationSpec generationSpec)
+            public Emitter(in OperationParameterInjectionSourceGenerationContext sourceGenerationContext, in SourceGenerationSpec generationSpec)
             {
                 _sourceGenerationContext = sourceGenerationContext;
                 _generationSpec = generationSpec;
+                _builder = new StringBuilder();
+                _builder.Clear();
```

Comment:
I'm trying to work out what benefit moving this to an instance field has? I think there might be a bug introduced here. The string builder isn't cleared before each call to EmitOperationContract. If _generationSpec.OperationContractSpecs contains more than one operationContractSpec, the second call to EmitOperationContract will have both sets of code accumulated in the string builder. The second and subsequent calls to _sourceGenerationContext.AddSource will add a file which contains all the prior generated code too.

Reply:
Thanks, there was indeed a bug - i added a unit test to confirm and fix in the latest commit i pushed.
The benefit is to tie the StringBuilder instance lifetime per generation pass instead of per file to generate.